lib/logstash/inputs/http_poller.rb

```diff
@@ -240,12 +240,20 @@ def request_async(queue, name, request)
 
   private
   def handle_success(queue, name, request, response, execution_time)
-    @codec.decode(response.body) do |decoded|
-      event = @target ? LogStash::Event.new(@target => decoded.to_hash) : decoded
+    body = response.body
+    # If there is a usable response. HEAD requests are `nil` and empty get
+    # responses come up as "" which the codec cannot deal with
+    if body && body.size > 0
+      @codec.decode(response.body) do |decoded|
+        event = @target ? LogStash::Event.new(@target => decoded.to_hash) : decoded
+        handle_decoded_event(queue, name, request, response, event, execution_time)
+      end
+    else
+      event = ::LogStash::Event.new
```

Comment:
I don't think the handling of 503s has changed at all. 503s can have bodies, so I don't think this impacts that use case.